File: src/types/model/withdraw.ts
export interface Withdraw {
  withdraw_id: number;
  card_number: string;
  withdraw_amount: number;
  withdraw_time: Date;
  created_at: string;
  updated_at: string;
}

export interface WithdrawTrashed {
  withdraw_id: number;
  card_number: string;
  withdraw_amount: number;
  withdraw_time: Date;
  created_at: string;
  updated_at: string;
  deleted_at: string;
}

export interface WithdrawMonthStatusSuccess {
  year: string;
  month: string;
  total_amount: number;
  total_success: number;
}

export interface WithdrawYearStatusSuccess {
  year: string;
  total_amount: number;
  total_success: number;
}

export interface WithdrawMonthStatusFailed {
  year: string;
  month: string;
  total_amount: number;
  total_failed: number;
}

export interface WithdrawYearStatusFailed {
  year: string;
  total_amount: number;
  total_failed: number;
}

export interface WithdrawMonthlyAmount {
  month: string;
  total_amount: number;
}

export interface WithdrawYearlyAmount {
  year: string;
  total_amount: number;
}

File: src-tauri/src/model/withdraw.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug)]
pub struct WithdrawResponse {
    pub withdraw_id: i32,
    pub withdraw_no: String,
    pub card_number: String,
    pub withdraw_amount: i32,
    pub withdraw_time: String,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Serialize, Deserialize)]
pub struct WithdrawResponseDeleteAt {
    pub withdraw_id: i32,
    pub withdraw_no: String,
    pub card_number: String,
    pub withdraw_amount: i32,
    pub withdraw_time: String,
    pub created_at: String,
    pub updated_at: String,
    pub deleted_at: String,
}

#[derive(Serialize, Deserialize)]
pub struct WithdrawResponseMonthStatusSuccess {
    pub year: String,
    pub month: String,
    pub total_amount: i32,
    pub total_success: i32,
}

#[derive(Serialize, Deserialize)]
pub struct WithdrawResponseYearStatusSuccess {
    pub year: String,
    pub total_amount: i32,
    pub total_success: i32,
}

#[derive(Serialize, Deserialize)]
pub struct WithdrawResponseMonthStatusFailed {
    pub year: String,
    pub total_amount: i32,
    pub month: String,
    pub total_failed: i32,
}

#[derive(Serialize, Deserialize)]
pub struct WithdrawResponseYearStatusFailed {
    pub year: String,
    pub total_amount: i32,
    pub total_failed: i32,
}

#[derive(Serialize, Deserialize)]
pub struct WithdrawMonthlyAmountResponse {
    pub month: String,
    pub total_amount: i32,
}

#[derive(Serialize, Deserialize)]
pub struct WithdrawYearlyAmountResponse {
    pub year: String,
    pub total_amount: i32,
}
